File: Information Network Security(INS)/lab/lab 2 xor/lab2final.java
public class lab2final {

    // Method to encrypt and decrypt text using Caesar Cipher
    public static String caesarCipher(String text, int shift, String mode) {
        StringBuilder result = new StringBuilder();
        
        if (mode.equalsIgnoreCase("decrypt")) {
            shift = -shift;  // For decryption, reverse the shift
        }

        // Iterate through each character in the text
        for (int i = 0; i < text.length(); i++) {
            char charAt = text.charAt(i);

            // Encrypt or decrypt alphabetic characters
            if (Character.isLetter(charAt)) {
                char base = Character.isUpperCase(charAt) ? 'A' : 'a';
                char newChar = (char) ((charAt - base + shift + 26) % 26 + base);
                result.append(newChar);
            } else {
                // Non-alphabetic characters remain unchanged
                result.append(charAt);
            }
        }

        return result.toString();
    }

    public static void main(String[] args) {
        String text = "Hello, World!";
        int shift = 3;

        // Encrypt the text
        String encryptedText = caesarCipher(text, shift, "encrypt");
        System.out.println("Encrypted Text: " + encryptedText);

        // Decrypt the text
        String decryptedText = caesarCipher(encryptedText, shift, "decrypt");
        System.out.println("Decrypted Text: " + decryptedText);
    }
}

File: Information Network Security(INS)/lab/lab 2 xor/lab2final.c
#include <stdio.h>
#include <ctype.h>

// Function to encrypt and decrypt text using Caesar Cipher
void caesarCipher(char* text, int shift, char* mode) {
    if (mode == "decrypt") {
        shift = -shift;  // Reverse the shift for decryption
    }

    // Iterate through each character in the string
    for (int i = 0; text[i] != '\0'; i++) {
        char charAt = text[i];

        // Encrypt or decrypt alphabetic characters
        if (isalpha(charAt)) {
            char base = isupper(charAt) ? 'A' : 'a';
            text[i] = (char)(((charAt - base + shift + 26) % 26) + base);
        }
    }
}

int main() {
    char text[] = "Hello, World!";
    int shift = 3;

    // Encrypt the text
    caesarCipher(text, shift, "encrypt");
    printf("Encrypted Text: %s\n", text);

    // Decrypt the text
    caesarCipher(text, shift, "decrypt");
    printf("Decrypted Text: %s\n", text);

    return 0;
}
